create-plugin-update/action.yml

```diff
@@ -0,0 +1,94 @@
+name: "Grafana Create Plugin Update"
+description: "Runs create-plugin update within a Grafana plugin and opens a PR with the changes"
+
+inputs:
+  token:
+    description: "Token for the repository. Can be passed in using `{{ secrets.GITHUB_TOKEN }}`."
+    required: false
+    default: "${{ github.token }}"
+  base:
+    description: "Sets the pull request base branch"
+    required: false
+    default: "main"
+  node-version:
+    description: "Version of node"
+    required: false
+    default: "20"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: "${{ inputs.node-version }}"
+
+    - name: Get latest version of create-plugin package
+      id: get-latest-version
+      run: echo "LATEST_VERSION=$(npx -y @grafana/create-plugin@latest version)" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+        ref: ${{ inputs.base }}
+
+    - name: Get version from .config/.cprc.json
+      id: get-config-version
+      run: |
+        CONFIG_VERSION=$(jq -r '.version' .config/.cprc.json)
+        echo "CONFIG_VERSION=${CONFIG_VERSION}" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Compare versions
+      id: compare-versions
+      run: |
+        if [ "$LATEST_VERSION" != "$CONFIG_VERSION" ]; then
+          echo "update_needed=true" >> $GITHUB_OUTPUT
+        else
+          echo "update_needed=false" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+
+    - name: Update the configs
+      if: steps.compare-versions.outputs.update_needed == 'true'
+      run: |
+        ${{ github.action_path }}/pm.sh update
+        ${{ github.action_path }}/pm.sh install
+      shell: bash
+
+    - name: Create branch
+      if: steps.compare-versions.outputs.update_needed == 'true'
+      run: |
+        git config --local user.name "github-actions[bot]"
+        git config --local user.email "github-actions[bot]@users.noreply.github.com"
+        git checkout -b update-grafana-create-plugin
+        git add .
+        git commit -m "chore: update configuration to create-plugin ${{ env.LATEST_VERSION }}"
+        git push --force https://x-access-token:${{ inputs.token }}@github.com/${{ github.repository }} update-grafana-create-plugin
```

Comment:
maybe not forcing this? In case there are changes already in the remote branch that needs care

Reply:
Hmm... you're right this needs a rethink. Right now it always works with the same branch name and the action I was using was force pushing to keep the single branch updated. This has worked well for my little plugin for the last 5 months or so but I see your point - it would be rather painful if a dev spent a load of time fixing up the PR then the next morning all their changes are gone and have to wrangle the branch.

Do you think each run that has an update should open a separate PR against a version specific branch (e.g. `update-grafana-create-plugin-${{ env.LATEST_VERSION }}`)? Like this the action would only ever push one commit to a specific branch and there wouldn't need to be force involved ever. Plugin devs could set it to run every month or so to keep noise down and they would be free to tackle any issues without the action causing them bother.

Reply:
There's a strong chance that right now a dev will want to do some manual intervention as the update command updates all the grafana npm packages to latest which isn't always ideal.

Reply:
> Do you think each run that has an update should open a separate PR against a version specific branch (e.g. `update-grafana-create-plugin-${{ env.LATEST_VERSION }}`)?

That sounds good (plus not forcing, in case the same GH action is executed twice).

Reply:
Added in https://github.com/grafana/plugin-actions/pull/38/commits/cabd6b49de3d2ad39f50bc5ecb692bc6f9e025b4